bindings/nodejs/lib/types/block/core/basic-block.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { Payload, PayloadDiscriminator } from '../payload';
+import { Type } from 'class-transformer';
+import { StrongParents, WeakParents, ShallowLikeParents } from '../parents';
+import { Block } from './block';
+
+/**
+ * Basic Block layout.
+ */
+export class BasicBlock extends Block {
```

Comment:
I don't know about that. This kinda changes the definition of `BasicBlock`

Reply:
Looks like a basically identical structure to the block enum could be created using TS generics and unions.

Reply:
So are you saying it would work or not?

Reply:
it works :)

Reply:
Not sure I understand what's being said here, but I don't really like the proposed version, it's not strictly implementing the TIP.

Reply:
Then please elaborate why not, i dont see whats missing

Reply:
Nothing is missing, but a `BasicBlock` shouldn't extend a `Block` and have access to the base fields, that's not how the TIP is. Why can't we use composition instead of this ?

Reply:
Because then the serialized structure becomes different.

instead of 
```
BasicBlock {
   Block wrapper fields...
   Basic Block fields...
}
```

it would be
```
BasicBlock {
   BlockWrapper {
      fields...
   },
   Basic Block fields...
}
```

Besides the tip says:
`The two block types introduced in this TIP share some common fields which are contained in an outer wrapper. This Block Wrapper is only introduced as a way to deduplicate the definition of those fields. It is not a standalone block type.`

What im doing is exactly that. (Base Block is abstract thus cannot instantiate)

Reply:
> it would be
> 
> ```
> BasicBlock {
>    BlockWrapper {
>       fields...
>    },
>    Basic Block fields...
> }
> ```

Surely this can be changed with a very simple custom JSON serialization function that removes the extra brackets, or another simple solution. I don't see this as a blocker.

Reply:
Yeah I don't know TS well enough to say, but I feel like the above isn't right anyway with composition. I'm pretty sure a union type like `BlockWrapper<BasicBlock> | BlockWrapper<ValidationBlock>` would work.

Reply:
rewrote it , we had a misunderstanding :=)